clients/filesystem-hadoop/src/main/java/com/datastrato/gravitino/filesystem/hadoop/GravitinoVirtualFileSystem.java

```diff
@@ -0,0 +1,385 @@
+      this.workingDirectory = newDir;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public Path getWorkingDirectory() {
+    return this.workingDirectory;
+  }
+
+  @Override
+  public boolean mkdirs(Path f, FsPermission permission) throws IOException {
+    NameIdentifier identifier = extractIdentifier(f.toUri());
+    FilesetMeta meta = getCachedFileset(identifier);
+    Path actualPath = resolvePathByIdentifier(identifier, meta, f);
+    return meta.getFileSystem().mkdirs(actualPath, permission);
+  }
+
+  @Override
+  public FileStatus getFileStatus(Path f) throws IOException {
+    NameIdentifier identifier = extractIdentifier(f.toUri());
+    FilesetMeta meta = getCachedFileset(identifier);
+    Path actualPath = resolvePathByIdentifier(identifier, meta, f);
+    FileStatus fileStatus = meta.getFileSystem().getFileStatus(actualPath);
+    return resolveFileStatusPathPrefix(
+        fileStatus, meta.getFileset().storageLocation(), concatFilesetPrefix(identifier, meta));
+  }
+
+  private String concatFilesetPrefix(NameIdentifier identifier, FilesetMeta meta) {
+    String filesetPrefix =
+        GravitinoVirtualFileSystemConfiguration.GVFS_FILESET_PREFIX
+            + identifier.namespace().level(1)
+            + "/"
+            + identifier.namespace().level(2)
+            + "/"
+            + identifier.name();
+    if (meta.getFileset().storageLocation().endsWith("/")) {
+      filesetPrefix += "/";
+    }
+    return filesetPrefix;
+  }
+
+  private Path resolvePathByIdentifier(NameIdentifier identifier, FilesetMeta meta, Path path) {
+    String originPath = path.toString();
+    if (!originPath.startsWith(GravitinoVirtualFileSystemConfiguration.GVFS_FILESET_PREFIX)) {
+      throw new InvalidPathException(
+          String.format(
+              "Path %s doesn't start with the scheme \"%s\"",
+              path, GravitinoVirtualFileSystemConfiguration.GVFS_FILESET_PREFIX));
+    }
+    try {
+      Path storageLocation = new Path(meta.getFileset().storageLocation());
+      boolean isMountSingleFile = meta.getFileSystem().isFile(storageLocation);
+      if (isMountSingleFile) {
+        Preconditions.checkArgument(
+            originPath.equals(concatFilesetPrefix(identifier, meta)),
+            "Cannot resolve path: %s to actual storage path, because the fileset only mounts a single file.",
+            path);
+        return storageLocation;
+      } else {
+        return new Path(
+            originPath.replaceFirst(
+                concatFilesetPrefix(identifier, meta), storageLocation.toString()));
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(
+          String.format("Cannot resolve path: %s to actual storage path, exception:", path), e);
+    }
+  }
+
+  private FileStatus resolveFileStatusPathPrefix(
+      FileStatus fileStatus, String fromPrefix, String toPrefix) {
+    String uri = fileStatus.getPath().toString();
```

Comment:
You'd better rename this variant as it will hide the attribute `uri` of this class.

Reply:
Done